connectors/sync_job_runner.py

```diff
@@ -227,14 +229,26 @@ async def prepare_docs(self):
             filtering=self.sync_job.filtering
         ):
             doc_id = str(doc.get("_id", ""))
-            doc_id_size = len(doc_id.encode("utf-8"))
+            doc_id_size = len(doc_id.encode(UTF_8))
 
             if doc_id_size > ES_ID_SIZE_LIMIT:
-                logger.error(
+                logger.debug(
                     f"Document with id '{doc_id}' with a size of '{doc_id_size}' bytes could not be ingested. "
-                    f"Elasticsearch has an upper limit of '{ES_ID_SIZE_LIMIT}' bytes for the '_id' field."
+                    f"Elasticsearch has an upper limit of '{ES_ID_SIZE_LIMIT}' bytes for the '_id' field.",
+                    "Hashing id...",
                 )
-                continue
+
+                hashed_id = self.source_klass.hash_id(doc_id)
```

Comment:
There's a default implementation in `BaseDataSource`, which simply returns the original `_id` without mutation. So no need to implement it explicitly in every data source.

Reply:
Since default implementation does nothing and most connectors don't need it, maybe it's worth to move this logic to generic sql database and use from there?

Reply:
I'm not too sure, whether other connectors won't need it in the future. I suspect we don't have a serious performance penalty with this indirection, right?

Reply:
I think there's barely any performance penalty, but added complexity seems really unnecessary here - it's potentially a problem when we have to generate pseudo-ids for the connectors AND these ids get too big. In time of Workplace Search I don't think there's any connector that hashes ids - it was not needed at all.

So I'd probably for now not make it a part of framework, mostly cause it does not seem to be useful for anything other than DBs that have no PK enforcement. How many are out there?

Reply:
I'm okay to merge it, but to me it just should never be in a framework itself unless it's used in most connectors or most connector authors would need it.

It might be just my opinion though, I prefer a bit more purist approach and providing overloadable methods make the framework harder to explain/understand.

Reply:
Late for the discussion.

The purpose of this PR is to convert an over-long ID to a new ID, which must be deterministic. What I meant, does it have to be provided by each connector implementation? Can the framework provide this functionality in `utils`?